core/src/main/java/hudson/model/Descriptor.java

```diff
@@ -802,8 +806,61 @@ public boolean configure( StaplerRequest req ) throws FormException {
      *      to keep the client in the same config page.
      */
     public boolean configure( StaplerRequest req, JSONObject json ) throws FormException {
-        // compatibility
-        return configure(req);
+
+        if (Util.isOverridden(Descriptor.class, getClass(), "configure", StaplerRequest.class)) {
+            // compatibility
+            return configure(req);
+        }
+
+        reset();
+        req.bindJSON(this, json);
+        save();
+        return true;
+    }
+
+    /**
+     * Reset descriptor data-bound attributes to default value.
+     *
+     * Properties are reset to their default values, calling the matchind DataBoundSetter method. Default value set is
+     * property type's default value, until a public static final field is declared with name
+     * <code>{{property_name}}_default</code>.
+     *
+     * Can overridden when this generic reset-to-defaults mechanism doesn't match with Descriptor's data model
+     * (in which case it is questionable if it shouldn't be fixed)
+     */
+    public void reset() {
+        for (PropertyDescriptor pd : PropertyUtils.getPropertyDescriptors(this)) {
+            final Method writeMethod = pd.getWriteMethod();
+            if (writeMethod != null && writeMethod.getAnnotation(DataBoundSetter.class) != null) {
+                try {
+                    final Class cl = writeMethod.getParameterTypes()[0];
+                    Object value;
+                    try {
+                        final Field field = getClass().getField(pd.getName() + "_default");
```

Comment:
Can't we create an instance and check what the getters return, similar to what the Pipeline Snippet Generator does?

Reply:
yes indeed, but relying on such convention allows to get the jelly tags `default` set as well.